src/transformations/io-transformations/PrintfToCoutTransformation.cpp

```diff
@@ -0,0 +1,224 @@
+}
+
+StringRef PrintfToCoutVisitor::getAsText(const Expr * e) {
+    SourceLocation LocStart = e->getBeginLoc();
+    SourceLocation LocEnd = Lexer::getLocForEndOfToken(LocStart, 0, sm, opt);
+    SourceRange range(LocStart, LocEnd);
+    return Lexer::getSourceText(CharSourceRange::getCharRange(range), sm, opt);
+}
+
+vector<string> PrintfToCoutVisitor::getPrecisionPrefix(const vector<string> &plholdersvec) {
+    vector<string> prefixlist;
+
+    string cur_mode;  // we only need to add fixed if fixed changed to scientific or vice versa
+    int cur_precision = -1;  // we only need to add setprecision if precision changed..
+
+
+    for (const string &plholder : plholdersvec) {
+        string cur_prefix;
+
+        // Check if we have s.th. like "%.10f"
+        auto floatmatches = getMatches(plholder, floatPrecisionRegex);
+        if (!floatmatches.empty()) {
+            if (floatmatches.size() != 1) {
+                errs() << "Warning: Format string has != 1 matches for %.Xf." << "\n";
+                throw IOTransformationException();
+            }
+            vector<string> splits = split(floatmatches[0], floatSplitRegex);
+            if (splits.size() != 2) {
+                errs() << "Error: Splits != 2.; size:"
+                       << splits.size() << "for:"
+                       << floatmatches[0] << "\n";
+                throw IOTransformationException();
+            } else {
+                if (cur_mode != "fixed") {
+                    cur_prefix += append_std("fixed");
+                    cur_mode = "fixed";
+                }
+                if (std::stoi(splits[1]) != cur_precision) {
+                    cur_precision = stoi(splits[1]);
+                    cur_prefix += " << " + append_std("setprecision(" + splits[1] + ") << ");
+                }
+            }
+        } else if (plholder == "%f") {
+            // Check if only %f, %e, %E,
+            // printf's default precision is 6.. that's why we use
+            // setprecision(6) in the default case %f
+            if (cur_mode != "fixed") {
+                cur_prefix += append_std("fixed") + " << ";
+                cur_mode = "fixed";
+            }
+            if (cur_precision != 6) {
+                cur_precision = 6;
+                cur_prefix += append_std("setprecision(6)") + " << ";
+            }
+
+        } else if (plholder == "%e" || plholder == "%E") {
+            if (cur_mode != "scientific") {
+                cur_prefix += append_std("scientific") + " << ";
+                cur_mode = "scientific";
+            }
+            if (cur_precision != 6) {
+                cur_precision = 6;
+                cur_prefix += append_std("setprecision(6)") + " << ";
+            }
+        }
+        prefixlist.push_back(cur_prefix);
+    }
+
+    if (prefixlist.size() != plholdersvec.size()) {
+        errs() << "placeholders and prefix list mismatch" << "\n";
```

Comment:
For me, this error doesn't say anything. What's the problem if their size mismatch?